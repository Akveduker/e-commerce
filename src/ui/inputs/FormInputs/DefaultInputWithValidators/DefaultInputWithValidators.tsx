import _ from 'lodash';
import { useInputsStateChangers } from '../../../../hooks/form/useInputsStateChangers';
import { FormDefaultInput } from '../../../../types/formInput/formInput';
import InputWithValidators from '../parts/InputWithValidators/InputWithValidators';
const DefaultInputWithValidators = <S extends string[]>(props: FormDefaultInput<S>) => {
    const { dataItemState, inputTitle, errorTitle, dispatch, stateItemKey, ...rest } = props
    const { status, value, errorBody } = dataItemState
    const [onChange, onBlur] = useInputsStateChangers<S>(_.pick(props, ['dispatch', 'stateItemKey', 'dataItemState']), value)
    return (
        <InputWithValidators
            inputTitle={inputTitle}
            errorTitle={errorTitle}
            status={status}
            errorBody={errorBody}
            onChange={onChange}
            onBlur={onBlur}
            value={value}
            {...rest}
        />
    )
};

export default DefaultInputWithValidators;