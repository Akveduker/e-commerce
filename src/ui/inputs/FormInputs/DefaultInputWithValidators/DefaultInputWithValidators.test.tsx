import { render, screen } from "@testing-library/react"
import DefaultInputWithValidators from "./DefaultInputWithValidators"

describe('DefaultInputWithValidators', () => {
    test('Только инпут без маски, снапшот', () => {
        const { container } = render(
            <DefaultInputWithValidators
                inputTitle="title"
                errorTitle="error"
                stateItemKey="phone"
                dispatch={jest.fn}
                dataItemState={{ errorBody: '', status: 'idle', value: '', validators: [] }}
            />
        )
        expect(screen.queryByTestId('default')).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })
})