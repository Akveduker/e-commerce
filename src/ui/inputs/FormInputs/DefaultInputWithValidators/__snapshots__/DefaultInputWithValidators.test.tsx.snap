// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultInputWithValidators Только инпут без маски, снапшот 1`] = `
<div>
  <label>
    <p
      class="name"
    >
      title
    </p>
    <label
      class="with-children container"
      data-testid="label"
    >
      <input
        class="with-children__input"
        data-testid="default"
        value=""
      />
      <div
        class="with-children__children"
      >
        <div />
      </div>
    </label>
    <div />
  </label>
</div>
`;
